Panic: nil pointer in copilot.initiatorTransport.RoundTrip — io.ReadAll(req.Body) when req.Body is nil

### Summary

Using a **GitHub Copilot** model crashes Crush with a nil-pointer panic in Crush's own Copilot OAuth transport. `internal/oauth/copilot/client.go`'s `initiatorTransport.RoundTrip` guards only against `req.Body == http.NoBody` before calling `io.ReadAll(req.Body)`, but a request can also have `req.Body == nil`. When it does, `io.ReadAll(nil)` dereferences a nil reader and the whole process panics (and, being on a background goroutine, takes the session down).

It reproduces via **title generation** (`sessionAgent.GenerateTitle`) and, from the stack, on a **retry** of the streaming request — which is consistent with the body having been consumed/reset to nil on the first attempt.

### Environment

- Crush **v0.81.0** (`charm.land/fantasy v0.34.0`, `charmbracelet/openai-go v0.0.0-20260617131321-5e4b9c18c4be`)
- Windows Server 2022 (platform-independent — the fault is in HTTP request handling)
- Provider: GitHub Copilot (authenticated via the in-app Copilot login)

### Panic (verbatim)

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal 0xc0000005 code=0x0 addr=0x18 pc=0x7ff6a9dd1285]

goroutine 161 [running]:
sync.(*WaitGroup).Go.func1.1()
        sync/waitgroup.go:251 +0x45
panic({0x7ff6abcaac60?, 0x7ff6aeb277d0?})
        runtime/panic.go:860 +0x13a
io.ReadAll({0x0, 0x0})
        io/io.go:722 +0xe5
github.com/charmbracelet/crush/internal/oauth/copilot.(*initiatorTransport).RoundTrip(...)
        github.com/charmbracelet/crush/internal/oauth/copilot/client.go:51 +0x2ac
net/http.send(...)
        net/http/client.go:264 +0x64b
net/http.(*Client).send(...)
        net/http/client.go:185 +0x258
net/http.(*Client).do(...)
        net/http/client.go:733 +0x9d7
net/http.(*Client).Do(...)
        net/http/client.go:592
github.com/charmbracelet/openai-go/internal/requestconfig.(*RequestConfig).Execute(...)
        internal/requestconfig/requestconfig.go:445 +0x857
github.com/charmbracelet/openai-go/internal/requestconfig.ExecuteNewRequest(...)
        internal/requestconfig/requestconfig.go:571 +0x9b
github.com/charmbracelet/openai-go.(*ChatCompletionService).NewStreaming(...)
        chatcompletion.go:107 +0x3cb
charm.land/fantasy/providers/openai.languageModel.Stream(...)
        charm.land/fantasy@v0.34.0/providers/openai/language_model.go:321 +0x36c
charm.land/fantasy.(*agent).Stream.func1()
        charm.land/fantasy@v0.34.0/agent.go:950 +0x159
charm.land/fantasy.retryWithExponentialBackoff[...](...)
        charm.land/fantasy@v0.34.0/retry.go:88 +0x12b
charm.land/fantasy.(*agent).Stream.RetryWithExponentialBackoffRespectingRetryHeaders[...].func2(...)
        charm.land/fantasy@v0.34.0/retry.go:60
charm.land/fantasy.(*agent).Stream(...)
        charm.land/fantasy@v0.34.0/agent.go:948 +0xf25
github.com/charmbracelet/crush/internal/agent.(*sessionAgent).GenerateTitle(...)
        github.com/charmbracelet/crush/internal/agent/agent.go:1732 +0x731
github.com/charmbracelet/crush/internal/agent.(*sessionAgent).Run.func1()
        github.com/charmbracelet/crush/internal/agent/agent.go:692 +0x32
sync.(*WaitGroup).Go.func1()
        sync/waitgroup.go:258 +0x4a
created by sync.(*WaitGroup).Go in goroutine 125
        sync/waitgroup.go:238 +0x73
```

### Root cause

`internal/oauth/copilot/client.go`:

```go
func (t *initiatorTransport) RoundTrip(req *http.Request) (*http.Response, error) {
    ...
    if req.Body == http.NoBody {          // <-- only handles http.NoBody
        req.Header.Set(xInitiatorHeader, userInitiator)
        return t.roundTrip(req)
    }

    req = req.Clone(req.Context())
    bodyBytes, err := io.ReadAll(req.Body) // <-- line 51: req.Body may be nil → panic
    ...
}
```

In Go's `net/http`, a bodyless request can have `req.Body == nil` **or** `req.Body == http.NoBody` — they are not the same value. This guard only catches `http.NoBody`, so a request whose `Body` is `nil` falls through to `io.ReadAll(req.Body)` → `io.ReadAll(nil)`, which panics on the nil reader (`io.ReadAll({0x0, 0x0})` in the trace).

The presence of `retryWithExponentialBackoff` in the stack suggests the trigger is a retry: the first attempt reads/consumes the body, and the retried `*http.Request` reaches `RoundTrip` with a `nil` Body.

### Suggested fix

Treat a `nil` body the same as `http.NoBody`:

```go
if req.Body == nil || req.Body == http.NoBody {
    req.Header.Set(xInitiatorHeader, userInitiator)
    return t.roundTrip(req)
}
```

(Optionally also make `GenerateTitle`'s background goroutine recover, so any future transport-layer panic can't take down the whole session — but the nil-body guard fixes this specific crash.)

### Steps to reproduce

1. Configure Crush with a GitHub Copilot model and sign in via the in-app Copilot login.
2. Start an interactive session and send a message.
3. Background title generation issues the request through the Copilot transport; on the code path where `req.Body` is `nil`, Crush panics with the trace above.

--- a/internal/oauth/copilot/client.go
+++ b/internal/oauth/copilot/client.go
@@ -37,8 +37,10 @@ func (t *initiatorTransport) RoundTrip(req *http.Request) (*http.Response, error
 	if req == nil {
 		return nil, fmt.Errorf("HTTP request is nil")
 	}
-	if req.Body == http.NoBody {
-		// No body to inspect; default to user.
+	if req.Body == nil || req.Body == http.NoBody {
+		// No body to inspect; default to user. A nil Body is valid for
+		// bodyless requests (e.g. GET), and is distinct from http.NoBody,
+		// so both must be handled before reading below.
 		req.Header.Set(xInitiatorHeader, userInitiator)
 		slog.Debug("Setting X-Initiator header to user (no request body)")
 		return t.roundTrip(req)

--- a/internal/oauth/copilot/client_test.go
+++ b/internal/oauth/copilot/client_test.go
@@ -51,11 +51,11 @@ func TestInitiatorTransportSetsHeader(t *testing.T) {
 			var err error
 			switch kind {
 			case 0:
-				req, err = http.NewRequest(http.MethodGet, srv.URL, nil)
+				req, err = http.NewRequestWithContext(t.Context(), http.MethodGet, srv.URL, nil)
 			case -1:
-				req, err = http.NewRequest(http.MethodGet, srv.URL, http.NoBody)
+				req, err = http.NewRequestWithContext(t.Context(), http.MethodGet, srv.URL, http.NoBody)
 			default:
-				req, err = http.NewRequest(http.MethodPost, srv.URL, strings.NewReader(payload))
+				req, err = http.NewRequestWithContext(t.Context(), http.MethodPost, srv.URL, strings.NewReader(payload))
 			}
 			require.NoError(t, err)
 

--- a/internal/oauth/copilot/client_test.go
+++ b/internal/oauth/copilot/client_test.go
@@ -1,0 +1,70 @@
+package copilot
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestInitiatorTransportSetsHeader(t *testing.T) {
+	tests := []struct {
+		name string
+		// body builds the request body; nil means a bodyless request.
+		body func() (int, string)
+		want string
+	}{
+		{
+			name: "nil body defaults to user",
+			body: func() (int, string) { return 0, "" }, // req.Body == nil
+			want: "user",
+		},
+		{
+			name: "NoBody defaults to user",
+			body: func() (int, string) { return -1, "" }, // req.Body == http.NoBody
+			want: "user",
+		},
+		{
+			name: "user-only history stays user",
+			body: func() (int, string) { return 1, `{"messages":[{"role":"user"}]}` },
+			want: "user",
+		},
+		{
+			name: "assistant history becomes agent",
+			body: func() (int, string) { return 1, `{"messages":[{"role":"assistant"}]}` },
+			want: "agent",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var got string
+			srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				got = r.Header.Get("X-Initiator")
+			}))
+			defer srv.Close()
+
+			kind, payload := tt.body()
+			var req *http.Request
+			var err error
+			switch kind {
+			case 0:
+				req, err = http.NewRequest(http.MethodGet, srv.URL, nil)
+			case -1:
+				req, err = http.NewRequest(http.MethodGet, srv.URL, http.NoBody)
+			default:
+				req, err = http.NewRequest(http.MethodPost, srv.URL, strings.NewReader(payload))
+			}
+			require.NoError(t, err)
+
+			client := &http.Client{Transport: &initiatorTransport{}}
+			resp, err := client.Do(req)
+			require.NoError(t, err)
+			resp.Body.Close()
+
+			require.Equal(t, tt.want, got)
+		})
+	}
+}

